lib/permutations.rb

```diff
@@ -1,4 +1,30 @@
 
 def permutations?(string1, string2)
+  return true if string1.empty? && string2.empty?
+  return false if string1.length != string2.length
   
+  hash = {}
+  
+  string_one = string1.split("")
+  
+  string_one.each do |letter|
+    if hash[letter]
+      hash[letter] += 1
+    else
+      hash[letter] = 1
+    end
+  end
+  
+  string_two = string2.split("")
+  
+  string_two.each do |letter|
+    if hash[letter]
+      hash[letter] -= 1
+    end
+    if hash[letter] == 0
+      hash.delete(letter)
+    end
+    
+  end
+  return hash.empty? ? true : false  
```

Comment:
Instead of deleting we could just set the count to 0 and then do `return hash.values.all? { |value| value == 0 }`

That way you don't have the expense of doing a delete.